File: app/src/main/java/com/mehul/redditwall/savedsub/SubViewModel.java
package com.mehul.redditwall.savedsub;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class SubViewModel extends AndroidViewModel {
    private SubRepository repo;
    private LiveData<List<SubSaved>> subs;

    public SubViewModel(@NonNull Application application) {
        super(application);
        repo = new SubRepository(application.getApplicationContext());
        subs = repo.getAllSubSaved();
    }

    public LiveData<List<SubSaved>> getAllSubs() {
        return subs;
    }

    public void deleteAll() {
        repo.deleteAll();
    }

    public void insert(SubSaved saved) {
        repo.insert(saved);
    }

    public void deleteSavedSub(SubSaved saved) {
        repo.deleteSubSaved(saved);
    }
}

File: app/src/main/java/com/mehul/redditwall/savedsub/SubViewModel.kt
package com.mehul.redditwall.savedsub

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

public class SubViewModel(application: Application) : AndroidViewModel(application) {
    private val repo: SubRepository = SubRepository(application.applicationContext)
    val allSubs: LiveData<List<SubSaved?>?>?

    fun deleteAll() {
        repo.deleteAll()
    }

    fun insert(saved: SubSaved?) {
        repo.insert(saved)
    }

    fun deleteSavedSub(saved: SubSaved?) {
        repo.deleteSubSaved(saved)
    }

    init {
        allSubs = repo.allSubSaved
    }
}